compiler/qsc_qasm/src/stdlib/QasmStd/src/QasmStd/Convert.qs

```diff
@@ -62,6 +65,38 @@ function ResultAsBigInt(input : Result) : BigInt {
     }
 }
 
+/// The ``RESULT_AS_DOUBLE`` function is used to implement the cast expr in QASM for result to float.
+function ResultAsDouble(value : Result) : Double {
+    if value == One {
+        1.
+    } else {
+        0.
+    }
+}
+
+/// The ``RESULT_ARRAY_AS_BOOL_BE`` function is used to implement the cast expr in QASM for bit[bits] to bool.
+/// with big-endian order. This is needed for round-trip conversion for bin ops.
+function ResultArrayAsBoolBE(array: Result[]) : Bool {
```

Comment:
Consider renaming this to just ResultArrayAsBool (without BE). And also, ResultArrayAsResult.
The result of the function doesn't depend on the order of elements. So, the BE suffix is just getting in the way.
There's a subtle difference in the order of evaluation - i.e. some results may be evaluated earlier, and some would not be evaluated at all. But I think this is just and implementation detail. Up to you.